### ISSUE TITLE
Proposed issue label for release management: release_blocker

### ISSUE THREAD

**Issue by michaelpacer** [jupyter/notebook#1890]
I want to suggest we have a new label for issues that are release blockers.

Premise: this tag should be only applied to issues that are considered genuine release blockers for the current release.

Danger: this tag has a potential uselessness property if it cannot be used to search only those issues which are blocking the current release. 

Solution: By definition of "release blocker" once the release is made, it is required that all release blockers will need to have been addressed. Therefore, we add one step (composed of 3 simple substeps) to our releasing instructions:

1. Search for all issues with the label "release_blocker".
2. Select all issues with the label "release_blocker".
3. Remove "release_blocker" label from selected issues. 

I propose that the current issue #1589 to be the first issue with which we test this tag and the first release with which we test this effort. 

A further modification could involve regularly following the following steps:

1. Search for all closed issues with the label "release_blocker" (is: issue, is: closed, label: release_blocker).
2. Select all closed issues with the label "release_blocker".
3. Remove "release_blocker" label from selected issues. 

Which would ensure at intermediate stages the label would only be used for those issues that are _still_ release breakers. 

Note: I'm fine with also manually removing the "release_blocker" label as they are completed. This just also proposes a release mechanism change that will ensure that if the semantics are followed and the mechanisms are used, this label will never be in an invalid state where it lists issues inappropriately as release breakers at release time, & will rarely be in an invalid state during the course of a release's development.

**Comment by takluyver** [comment]
Interesting idea, but I'm not convinced that we need to add anything to our process.

We already use milestones to indicate issues which we'd like to fix before a given release - e.g. [here's the 4.3 milestone](https://github.com/jupyter/notebook/milestone/9). When preparing for a release, we go through those issues to either fix them or bump them to a later milestone. I've just done a pass on this for 4.3. In theory, we could save some time if we had an easy way to distinguish issues we consider important to fix from things that are less important, but we'd put the fix in that release if one was ready. But I suspect it wouldn't save that much time, and I think going back and looking at these issues again is probably healthy for the project.

If we do add a label, I'd argue that it should be the other way round, i.e. tag issues that we don't mind bumping to a later release. If we tag release blockers, we would still need to go through the issues without the tag to see if any of them should get the tag. I.e. if we add an issue to a milestone, I'd treat it as a release blocker by default.

**Comment by minrk** [comment]
I agree with Thomas here. There is already a requirement that all milestoned issues are resolved or explicitly postponed to another milestone before we can release. Adding a 'blocker' tag doesn't seem to improve that process.

**Comment by ellisonbg** [comment]
I am doing a bunch of issue/label triage right now and have removed the release_blocker label as it embeds the same information as the milestones. I am also converting the labels to use the taxonomy of JuptyerLab to help us in the transition.
